codebuild/bin/grep_simple_mistakes.sh

```diff
@@ -164,6 +164,19 @@ for dir in $S2N_ERROR_IF_FREE; do
 done
 
 #############################################
+
+#############################################
+## Assert all ".[c|h]" source files have the correct file mode.
+#############################################
+S2N_FILES_WITH_WRONG_MODE=$(find . \( -name "*.c" -o -name "*.h" \) -a ! -perm 644)
```

Comment:
It's generally a good idea to do `\!`, unless shells have gotten significantly better recently :) I've also heard you can do single quotes. From the man page: "This character will also usually need protection from interpretation by the shell."
```suggestion
#############################################
S2N_FILES_WITH_WRONG_MODE=$(find . \( -name "*.c" -o -name "*.h" \) -a \! -perm 644)
```